addons/udes_sale_stock/models/partner_pii.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+
+# @TODO: EDI, testing, tidy, forms field read-only
+
+from odoo import models, fields, api, SUPERUSER_ID, _
+import logging
+
+_logger = logging.getLogger(__name__)
+
+REDACTED = _('Redacted')
+
+PII_RELATIONS = [
+    'partner_id', 'parent_id', 'res_partner_id', 'author_id', 'partner_address_id', 'owner_id', 
+    'partner_shipping_id', 'partner_invoice_id', 'order_partner_id', 'customer_id'
+]
+
+
+@api.model
+def _read(self, result):
+    """Redact fields based on res.user.u_view_customers"""
+
+    if self.env.uid == SUPERUSER_ID or self.env.user.u_view_customers:
+        return result
+
+    for record in result:          
+        for field in record:
+            if field in PII_RELATIONS and isinstance(record[field], tuple) and \
+                self.env['res.partner'].sudo().search_count([('id', '=', record[field][0]),('customer', '=', True)]):
```

Comment:
Too long. Besides, this bit of the if clause is aligned with the actual block which makes it confusing.

Reply:
Also, need space between tuples.